Userland/Utilities/checksum.cpp

```diff
@@ -61,10 +65,68 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
             }
             file = file_or_error.release_value();
         }
+        if (!verify_from_paths) {
+            while (!file->eof() && !file->has_error())
+                hash.update(file->read(PAGE_SIZE));
+            outln("{:hex-dump}  {}", hash.digest().bytes(), path);
+        } else {
+            StringBuilder buffer;
+            while (!file->eof() && !file->has_error())
+                buffer.append(file->read(PAGE_SIZE));
+            Vector<StringView> lines = buffer.string_view().split_view("\n"sv);
+            if (lines.size() == 0) {
+                ++read_fail_count;
+                continue;
+            }
+            for (size_t i = 0; i < lines.size(); ++i) {
+                Vector<StringView> line = lines[i].split_view("  "sv);
+                if (line.size() != 2) {
+                    ++read_fail_count;
+                    continue;
+                }
+
+                // line[i] = checksum
+                // line[i + 1] = filename
+                for (size_t i = 0; i < line.size(); i += 2) {
+                    StringView filename = line[i + 1];
+                    auto file_or_error = Core::File::open(filename, Core::OpenMode::ReadOnly);
+                    if (file_or_error.is_error()) {
+                        warnln("{}: {}: {}", program_name, filename, file->error_string());
+                        ++read_fail_count;
+                        ;
+                        continue;
+                    }
+                    NonnullRefPtr<Core::File> filename_from_file = file_or_error.release_value();
```

Comment:
I don't think that the name is appropriate, `file_from_filename` is probably a bit better.